contracts/src/servers/types/types.ts

```diff
@@ -0,0 +1,32 @@
+export interface ContractState {
+  owner: string;
+  evolve?: string;
+  servers: {
+    [serverId: string]: {
+      serverName: string;
+      contractTxId: string;
+    };
+  };
+}
+
+export interface ContractAction {
+  input: ContractInput;
+  caller: string;
+}
+
+export type ContractReadResult = {
+  serverId: string;
+  serverName: string | null;
```

Comment:
I wanted to explicitely return null when reading contract state and e.g. server is not found.

This is how the result of the read function code looks like:
```
return {
    result: { serverId, serverName: server?.serverName || null, contractTxId: server?.contractTxId || null },
  }
  ```
  
 I can change it to `serverName?: string` and `undefined` will be returned instead of `null`.